src/node_url.cc

```diff
@@ -1289,22 +1290,31 @@ namespace url {
         argv[ARG_PORT] = Integer::New(isolate, url.port);
       if (DOES_HAVE_PATH(url))
         argv[ARG_PATH] = Copy(env, url.path);
+      (void)cb->Call(context, recv, 9, argv);
+    } else if (error_cb->IsFunction()) {
+      Local<Value> argv[2] = { undef, undef };
+      argv[ERR_ARG_FLAGS] = Integer::NewFromUnsigned(isolate, url.flags);
+      argv[ERR_ARG_INPUT] =
+        String::NewFromUtf8(env->isolate(),
+                            input,
+                            v8::NewStringType::kNormal).ToLocalChecked();
+      (void)error_cb.As<Function>()->Call(context, recv, 2, argv);
```

Comment:
I know it was there already, but using `arraysize(argv)` should prevent the need for a hardcoded array length.